In [1]:
import pandas as pd
import numpy as np
import json
import nltk
import re
import csv
import matplotlib.pyplot as plt
from tqdm import tqdm
from sklearn.feature_extraction.text import TfidfVectorizer
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import MultiLabelBinarizer
from sklearn.linear_model import LogisticRegression
from sklearn.multiclass import OneVsRestClassifier
from sklearn.metrics import f1_score, accuracy_score
from data import * 



**Computational Tool Final Project**


In [2]:
data()


16559it [00:00, 19073.87it/s]
100%|██████████| 16559/16559 [00:00<00:00, 1039536.01it/s]


book_id                                          book_name  \
0           620                                        Animal Farm   
1           843                                 A Clockwork Orange   
2           986                                         The Plague   
3          1756          An Enquiry Concerning Human Understanding   
4          2080                               A Fire Upon the Deep   
...         ...                                                ...   
16554  36934824                                     Under Wildwood   
16555  37054020                                  Transfer of Power   
16556  37122323                                            Decoded   
16557  37132319  America Again: Re-becoming The Greatness We Ne...   
16558  37159503                                          Poor Folk   

                                                   genre  \
0      {"/m/016lj8": "Roman \u00e0 clef", "/m/06nbt":...   
1      {"/m/06n90": "Science Fiction", "/m/0l67h": "N...   
2      {"/m/02m4t": "Existentialism", "/m/02xlf": "Fi...   
3                                                          
4      {"/m/03lrw": "Hard science fiction", "/m/06n90...   
...                                                  ...   
16554                                                      
16555   {"/m/01jfsb": "Thriller", "/m/02xlf": "Fiction"}   
16556                       {"/m/0xdf": "Autobiography"}   
16557                                                      
16558  {"/m/02ql9": "Epistolary novel", "/m/014dfn": ...   

                                                 summary  
0       Old Major, the old boar on the Manor Farm, ca...  
1       Alex, a teenager living in near-future Englan...  
2       The text of The Plague is divided into five p...  
3       The argument of the Enquiry proceeds by a ser...  
4       The novel posits that space around the Milky ...  
...                                                  ...  
16554   Prue McKeel, having rescued her brother from ...  
16555   The reader first meets Rapp while he is doing...  
16556   The book follows very rough chronological ord...  
16557   Colbert addresses topics including Wall Stree...  
16558   Makar Devushkin and Varvara Dobroselova are s...  

[16559 rows x 4 columns]

On this project we will  use this data set(https://www.kaggle.com/code/iamhungundji/book-summary-genre-prediction/data). The dataset contain 16559 books with 4 columns 
* id 
* book name 
* genre 
* summary


Our main goal is to classify the books with text simmilarity ( using summaries) and predict their genre correctly. Moreover, we want to make a simple recomendation UI using github pages which the user can choose between time, rating , genre and get some recomendation based on our classification model.
